Update useSelect calls to useInViewSelect

tofumatt opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses. -->
Addresses issue #4121.

## Relevant technical choices

To reduce the manual import/changes across the repo, this exports `useInViewSelect` from `googlesitekit-data`.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

eugene-manuilov reviewed (changes requested):
Awesome work, @tofumatt! This almost looks good to me. Added a few questions, plus looks like some storybook stories are broken, do you mind checking it?

Some of broken stories:

* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/analytics-module-components-dashboard-popular-pages-widget--loaded
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/analytics-module-components-dashboard-popular-pages-widget--data-unavailable
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/analytics-module-components-dashboard-popular-pages-widget--zero-data
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/adsense-module-components-dashboard-top-earning-pages-widget--loaded
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/adsense-module-components-dashboard-top-earning-pages-widget--error
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/adsense-module-components-dashboard-top-earning-pages-widget--data-unavailable
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/adsense-module-components-module-top-earning-pages-widget--loaded
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/adsense-module-components-module-top-earning-pages-widget--data-unavailable
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/adsense-module-components-module-top-earning-pages-widget--error
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/adsense-module-components-module-top-earning-pages-widget--ad-sense-not-linked
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/adsense-module-components-module-overview-widget--error
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/analytics-module-components-page-dashboard-bounce-rate-widget--loaded
* https://google.github.io/site-kit-wp/storybook/pull/4408/?path=/story/analytics-module-components-module-page-popular-pages-widget--loaded

tofumatt commented:
Thanks for catching the broken stories! I thought our CI checks verified none were broken but maybe that's just some of them being scanned? Much obliged 👍🏻

tofumatt commented on assets/js/components/Root/index.js:
```diff
@@ -36,21 +41,25 @@ import { Provider as ViewContextProvider } from './ViewContextContext';
 import InViewProvider from '../InViewProvider';
 
 export default function Root( { children, registry, viewContext = null } ) {
+	const inViewState = useRef( {
```
Only because this one never changes, so I didn't think it was needed to create the state variable. But since we use state for this elsewhere: happy to do so 👍🏻

tofumatt commented on assets/js/components/adminbar/AdminBarClicks.js:
```diff
@@ -37,10 +37,10 @@ import PreviewBlock from '../PreviewBlock';
 import sumObjectListValue from '../../util/sum-object-list-value';
 import { partitionReport } from '../../util/partition-report';
-const { useSelect } = Data;
+const { useSelect, useInViewSelect } = Data;
```
We _technically_ don't, but I thought it was good practice and also a good test to make sure the `InViewProvider` was working everywhere in the app.

I can revert it, I guess there's no need for it but I figured it didn't hurt and was best practice throughout the code. 🤷🏻‍♂️

tofumatt commented:
Looks like a few of these stories are broken in `develop`:

* https://google.github.io/site-kit-wp/storybook/develop/?path=/story/adsense-module-components-module-top-earning-pages-widget--loaded
* https://google.github.io/site-kit-wp/storybook/develop/?path=/story/analytics-module-components-module-page-popular-pages-widget--data-unavailable

I took a quick look but I'm not sure why; I find these `generateReportBasedWidgetStories` functions really hard to debug/follow, so maybe we should do this in a separate PR 😓 

I fixed the rest of them, removed the `useInViewSelect` from the Admin Bar, and restored the WPComponent import for JSDoc, so I think this is good-to-go now.

eugene-manuilov reviewed (approved):
Thanks, @tofumatt. All seems to work good now. Hope we won't break anything after merging this into develop 🤣 🤞

eugene-manuilov reviewed (changes requested):
Ok, there is a merge conflict. Could you please fix it?

tofumatt commented:
All good; it was a very minor merge conflict, so fixed up and should be good-to-merge now 🙂

eugene-manuilov reviewed (approved):
Awesome. Thanks, @tofumatt.
